src/neuroconv/tools/data_transfers/_aws.py

```diff
@@ -30,3 +36,289 @@ def estimate_s3_conversion_cost(
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+
+    # It is very useful to have a status tracker that is separate from the job environment
+    # Technically detailed logs of inner workings are given in the CloudWatch, but that can only really be
+    # analyzed from the AWS web console
+    current_tables = dynamodb_client.list_tables()["TableNames"]
+    if status_tracker_table_name not in current_tables:
+        table = dynamodb_resource.create_table(
+            TableName=status_tracker_table_name,
+            KeySchema=[dict(AttributeName="id", KeyType="HASH")],
+            AttributeDefinitions=[dict(AttributeName="id", AttributeType="S")],
+            ProvisionedThroughput=dict(ReadCapacityUnits=1, WriteCapacityUnits=1),
+        )
+    else:
+        table = dynamodb_resource.Table(name=status_tracker_table_name)
+
+    # Ensure role policy is set
+    current_roles = [role["RoleName"] for role in iam_client.list_roles()["Roles"]]
+    if iam_role_name not in current_roles:
+        assume_role_policy = dict(
+            Version="2012-10-17",
+            Statement=[
+                dict(Effect="Allow", Principal=dict(Service="ecs-tasks.amazonaws.com"), Action="sts:AssumeRole"),
+            ],
+        )
+
+        role = iam_client.create_role(RoleName=iam_role_name, AssumeRolePolicyDocument=json.dumps(assume_role_policy))
+        iam_client.attach_role_policy(
+            RoleName=role["Role"]["RoleName"], PolicyArn="arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess"
+        )
+        iam_client.attach_role_policy(
+            RoleName=role["Role"]["RoleName"], PolicyArn="arn:aws:iam::aws:policy/CloudWatchFullAccess"
+        )
+    else:
+        role = iam_client.get_role(RoleName=iam_role_name)
+
+    # Ensure compute environment is setup
+    current_compute_environments = [
+        environment["computeEnvironmentName"]
+        for environment in batch_client.describe_compute_environments()["computeEnvironments"]
+    ]
+    if compute_environment_name not in current_compute_environments:
+        batch_client.create_compute_environment(
+            computeEnvironmentName=compute_environment_name,
+            type="MANAGED",
+            state="ENABLED",
+            computeResources={
+                "type": "EC2",
+                "allocationStrategy": "BEST_FIT",
+                "minvCpus": 0,
+                "maxvCpus": 256,
+                "subnets": ["subnet-0be50d51", "subnet-3fd16f77", "subnet-0092132b"],
+                "instanceRole": "ecsInstanceRole",
+                "securityGroupIds": ["sg-851667c7"],
+                "instanceTypes": ["optimal"],
+            },
+        )
+
+    # Ensure job queue exists
+    current_job_queues = [queue["jobQueueName"] for queue in batch_client.describe_job_queues()["jobQueues"]]
+    if job_queue_name not in current_job_queues:
+        batch_client.create_job_queue(
```

Comment:
TODO: expose argument for job state timeouts (when in 'runnable' not 'running' stalls out due to incompatibility of resource specification combined with spot %

Reply:
Looking further into this, it seems all timeout conditions are for the event that the job exits `RUNNABLE` state and enters `RUNNING`. How long it stays in that state is entirely up to how much data we expect to process and how fast we expect it to be processed

So I'm OK with setting a somewhat large value here just for the sake of safety, but it doesn't solve the problem of jobs not deploying (or the problem of how to track that easily from our tools)